lib/metanorma/site_manifest.rb

```diff
@@ -6,6 +6,21 @@ module Metanorma
   module SiteManifest
     class SourceEntries < Lutaml::Model::Serializable
       attribute :files, :string, collection: true
+
+      key_value do
+        map "files", to: :files, value_map: {
```

Comment:
@ribose-jeffreylau I don't see the reason to change to a non-default behavior for this key. Why don't we just fix the relevant code and the specs?

Reply:
The reason to change to a default value of empty array is it has been the expected behaviour before the change to using lutaml-model:

https://github.com/metanorma/metanorma-cli/blob/77bbccdd8d90f4250d2127938df5858f57da3550/lib/metanorma/cli/site_generator.rb#L126-L129

... and before the upgrade to lutaml-model 0.7:

https://github.com/metanorma/metanorma-cli/blob/8fdec680eedb022ea713db61a6a44c0ab484c108/spec/metanorma/site_manifest_spec.rb#L44

Reply:
Then please fix up the old expectations. The old expectation is based on faulty logic...

Reply:
I also don't understand why we are not using the lutaml-model representation of metanorma.yml in site_generator.rb.

Reply:
> I also don't understand why we are not using the lutaml-model representation of metanorma.yml in site_generator.rb.

This is the lutaml-model representation of metanorma.yml in the current site_generator.rb:

https://github.com/metanorma/metanorma-cli/blob/8fdec680eedb022ea713db61a6a44c0ab484c108/lib/metanorma/cli/site_generator.rb#L169-L178